mantle/platform/qemu.go

```diff
@@ -658,10 +660,20 @@ func newGuestfish(diskImagePath string, diskSectorSize int) (*coreosGuestfish, e
 	guestfishArgs = append(guestfishArgs, "-a", diskImagePath)
 	cmd := exec.Command("guestfish", guestfishArgs...)
 	cmd.Env = append(os.Environ(), "LIBGUESTFS_BACKEND=direct")
+
+	// Hack to run with a wrapper on older P8 hardware running RHEL7
 	switch system.RpmArch() {
 	case "ppc64le":
-		cmd.Env = append(os.Environ(), "LIBGUESTFS_HV=/usr/lib/coreos-assembler/libguestfs-ppc64le-wrapper.sh")
+		u := unix.Utsname{}
+		if err := unix.Uname(&u); err != nil {
+			return nil, errors.Wrapf(err, "detecting kernel information")
+		}
+		if strings.Contains(fmt.Sprintf("%s", u.Release), "el7") {
```

Comment:
I think this would be slightly better as:

```suggestion
		if strings.Contains(string(u.Release), "el7") {
```